Realm/Tests/MigrationTests.mm

```diff
@@ -1133,21 +1133,29 @@ - (void)testEnumerateObjectsAfterDeleteDataForRemovedType {
         [IntObject createInRealm:realm withValue:@[@2]];
     }];
 
-    RLMRealmConfiguration *config = self.config;
-    config.objectClasses = @[StringObject.class];
-    config.schemaVersion = 1;
-    config.migrationBlock = ^(RLMMigration *migration, uint64_t) {
+    RLMRealmConfiguration *config1 = self.config;
+    config1.objectClasses = @[StringObject.class];
+    config1.schemaVersion = 1;
+    config1.migrationBlock = ^(RLMMigration *migration, uint64_t) {
         [migration enumerateObjects:IntObject.className block:^(RLMObject *oldObject, RLMObject *newObject) {
+            // Object must still be accessible during the migration it gets deleted in.
             XCTAssertNotNil(oldObject);
+            XCTAssert([oldObject[@"intCol"] intValue] > 0);
             XCTAssertNil(newObject);
         }];
         [migration deleteDataForClassName:IntObject.className];
+    };
+    XCTAssertTrue([RLMRealm performMigrationForConfiguration:config1 error:nil]);
+    
+    RLMRealmConfiguration *config2 = self.config;
+    config2.objectClasses = @[StringObject.class];
+    config2.schemaVersion = 2;
+    config2.migrationBlock = ^(RLMMigration *migration, uint64_t) {
         [migration enumerateObjects:IntObject.className block:^(RLMObject *, RLMObject *) {
```

Comment:
Should add an assertion inside the migration block to ensure that it's executed

Reply:
There is an `XCTFail(@"should not have enumerated any objects");` in the next line. This block is supposed to **not** be executed since `IntObject` was already deleted one version earlier.

Reply:
Yes, but the migration block is meant to be called. My comment should have been one line higher

Reply:
Oh, true! Good point. Adding a check there too. 👍